app/src/main/java/org/mozilla/fenix/home/blocklist/BlocklistHandler.kt

```diff
@@ -51,6 +52,32 @@ class BlocklistHandler(private val settings: Settings) {
             }
         }
 
+    /**
+     * If the state is set to [RecentSyncedTabState.Success], filter the list of recently synced
+     * tabs by the blocklist. If the filtered list of tabs is empty, change the state to
+     * [RecentSyncedTabState.None]
+     *
+     * @param blocklistHandler The blocklist handler used to filter the recent synced tabs.
+     */
+    @JvmName("filterRecentSyncedTabState")
+    internal fun RecentSyncedTabState.filteredByBlocklist(blocklistHandler: BlocklistHandler) =
```

Comment:
Now that this function has been moved into this class, the `blocklistHandler` parameter can be removed. Defining the extension function within the class requires a class instance to be in scope to call the function. This can be more clearly demonstrated by checking that `this@filteredByBlocklist` and `this@BlocklistHandler` are both in scope in this function. 

That's why all the `filteredByBlocklist` calls are in `with (blocklistHandler)` blocks. I think doing so here will help keep this in convention with the rest of the file.

This is a stylistic thing I've picked up from looking into the source of Jetpack Compose. For example, some `Modifier` extension functions require certain scopes like `RowScope`. [Here is an example](https://developer.android.com/reference/kotlin/androidx/compose/foundation/layout/RowScope#(androidx.compose.ui.Modifier).align(androidx.compose.ui.Alignment.Vertical)) 

Kotlin 1.6.20 introduced a prototype of [context receivers](https://kotlinlang.org/docs/whatsnew1620.html#prototype-of-context-receivers-for-kotlin-jvm) that will allow us to use this pattern outside of the actual class definitions as well. Then this same pattern can be applied to code we don't own like library code.

Reply:
Thanks for the thorough explanation!